### PR TITLE
Add empty string check to Locale toString

### PR DESCRIPTION
**Pull request by GaryQian**
Fixes https://github.com/flutter/flutter/issues/43607

### CODE REVIEW

File: lib/ui/window.dart [hunk 1]
```diff
@@ -508,9 +508,9 @@ class Locale {
 
   String _rawToString(String separator) {
     final StringBuffer out = StringBuffer(languageCode);
-    if (scriptCode != null)
+    if (scriptCode != null && scriptCode != '')
```
Comment (jmagman) [inline lib/ui/window.dart]: isNotEmpty?

Comment (flar) [reply lib/ui/window.dart]: There are other lines of code in this file that also compare to `''`...

**Comment by flar** [comment]
LGTM

**Comment by GaryQian** [comment]
Going to land this through the fuchisa LUCI failures, as this has nothing to do with it and the infra may see logistical issues in the near future due to WFH/quarantine

**Pull request merged**